File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-13.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        i = 0
        j = n - 1

        while i <= j:
            mid = (i + j) // 2
            if nums[mid] == target:
                return mid

            if nums[i] == target:
                return i
            
            if nums[j] == target:
                return j


            if target < nums[mid] and target > nums[i] and nums[mid] > nums[i]:
                j = mid - 1
            elif target < nums[mid] and target < nums[i] and nums[mid] < nums[i]:
                j = mid - 1
            else:
                i = mid + 1


        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-13.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> my_map;

        for (int i=0; i<nums.size(); i++) {
            int find = target - nums[i];
            if (my_map[find] > 0) {
                return {my_map[find]-1, i};
            }

            my_map[nums[i]] = i+1;
        }

        return {};
    }
};
